lib/src/layout_grid_controller.dart

```diff
@@ -54,29 +54,22 @@ class LayoutGridController extends ChangeNotifier {
     _visibleGrid = !_visibleGrid;
     notifyListeners();
   }
-
-  static LayoutGridController of(BuildContext context) {
-    final controller = context
-        .dependOnInheritedWidgetOfExactType<LayoutGridScope>()
-        ?.controller;
-    if (controller != null) {
-      return controller;
-    }
-    throw UnimplementedError();
-  }
 }
 
-class LayoutGridScope extends InheritedWidget {
-  const LayoutGridScope({
+class LayoutGridController extends InheritedNotifier<LayoutGridNotifier> {
+  const LayoutGridController({
     super.key,
-    required this.controller,
+    required LayoutGridNotifier super.notifier,
     required super.child,
   });
 
-  final LayoutGridController controller;
-
-  @override
-  bool updateShouldNotify(LayoutGridScope oldWidget) {
-    return controller != oldWidget.controller;
+  static LayoutGridNotifier of(BuildContext context) {
+    final notifier = context
+        .dependOnInheritedWidgetOfExactType<LayoutGridController>()
+        ?.notifier;
+    if (notifier != null) {
+      return notifier;
+    }
+    throw UnimplementedError();
```

Comment:
this suggests unfinished library, in my opinion we should use more descriptive exception here